### PR TITLE
AB#9145 Assessment Scoring Fields

### PR DESCRIPTION
**Pull request by aurelio-aot**

### CODE REVIEW

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/Assessments/AssessmentAppService.cs [hunk 1]
```diff
@@ -188,5 +189,38 @@ private static OperationAuthorizationRequirement GetActionAuthorizationRequireme
...
         #endregion ASSESSMENT WORKFLOW
+
+        public async Task UpdateAssessmentScore(AssessmentScoresDto dto)
+        {
+            try
+            {
+                var assessment = await _assessmentRepository.GetAsync(dto.AssessmentId);
+                if (assessment != null)
+                {
+                    if(CurrentUser.GetId() != assessment.AssessorId)
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Application/Assessments/AssessmentAppService.cs:200]: These are business rules that need to be moved into the domain layer, and possibly the state machine

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/Assessments/AssessmentAppService.cs [hunk 2]
```diff
@@ -188,5 +189,38 @@ private static OperationAuthorizationRequirement GetActionAuthorizationRequireme
...
+                    {
+                        throw new AbpValidationException("Error: You do not own this assessment record.");
+                    }
+                    if (assessment.Status.Equals(AssessmentState.COMPLETED))
+                    {
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Application/Assessments/AssessmentAppService.cs:205]: Business rule

**Review by AndreGAot — COMMENTED** [review]
Some unit tests could be added for this too.

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/Assessments/AssessmentAppService.cs (hunk 1 above)
Comment (aurelio-aot) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/Assessments/AssessmentAppService.cs:200]: > These are business rules that need to be moved into the domain layer, and possibly the state machine

hi @AndreGAot -- can i remove this logic for now and rework on it in a separate user story for "assessment score state machine"?

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/Assessments/AssessmentAppService.cs (hunk 2 above)
Comment (aurelio-aot) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/Assessments/AssessmentAppService.cs:205]: > Business rule

same as above

Comment (AndreGAot) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/Assessments/AssessmentAppService.cs:210]: As discussed with @aurelio-aot  - will resolve with upcoming dynamic scoring feature

**Comment by aurelio-aot** [comment]
> Some unit tests could be added for this too.

hi @AndreGAot -- i have now pushed the unit test for this ticket. thanks.

**Review by AndreGAot — APPROVED** [review]
Approved for Dev

**Pull request merged**